src/osd/ReplicatedPG.cc

```diff
@@ -8311,6 +8359,28 @@ class C_OSD_RepopCommit : public Context {
   }
 };
 
+class C_OSD_RepopCommit_no_pg_lock : public Context {
+  ReplicatedPGRef pg;
+  boost::intrusive_ptr<ReplicatedPG::RepGather> repop;
+  ReplicatedPG::RepGather * real_repop;
```

Comment:
Managing the same resource both with a raw and a "smart" pointer is very error prone